### PR TITLE
Basic lambda wrappers

### PR DESCRIPTION
**Pull request by gianlucadecola**
# Description

This PR provides basic lambda wrappers which apply a function to the observation, actions and reward.
Also a wrapper for reward clipping is added.

### CODE REVIEW

File: gymnasium/dev_wrappers/lambda_action.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+from gymnasium.dev_wrappers import ArgType
+
+
+class LambdaActionV0(gymnasium.ActionWrapper):
+    """A wrapper that provides a function to modify the action passed to :meth:`step`.
+
+    Example to convert continuous actions to discrete:
+        >>> import gymnasium as gym
+        >>> import numpy as np
+        >>> from gymnasium.spaces import Dict
```
Comment (pseudo-rnd-thoughts) [inline gymnasium/dev_wrappers/lambda_action.py:16]: Remove

File: gymnasium/dev_wrappers/lambda_observations.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+from gymnasium.core import ObsType
+from gymnasium.dev_wrappers import ArgType
+
+
+class LambdaObservationsV0(gymnasium.ObservationWrapper):
+    """Lambda observation wrapper where a function is provided that is applied to the observation.
+
+    Example:
+        >>> import gymnasium as gym
+        >>> from gymnasium.spaces import Dict, Discrete
```
Comment (pseudo-rnd-thoughts) [inline gymnasium/dev_wrappers/lambda_observations.py:15]: Remove

File: tests/dev_wrappers/utils.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+import numpy as np
+
+import gymnasium
+from gymnasium import Space
+from gymnasium.core import ActType, ObsType
+from gymnasium.spaces import Box, Discrete
+
+
+class TestingEnv(gymnasium.Env):
```
Comment (pseudo-rnd-thoughts) [inline tests/dev_wrappers/utils.py:12]: This already exists in `tests/testing_env.py`, could you modify that implementation instead of this

**Review by pseudo-rnd-thoughts — CHANGES_REQUESTED** [review]
Could you remove the TestingEnv, otherwise looks good from the discussions we had with Ariel. Two small points for going forward
1. Don't worry testing vector implementations due to `VectorEnv` changes these will most likely have to be removed in the future. 
2. There are a couple of type hinting things we can improve (nothing breaking) however in an effect to merge this quickly I can do them after.

**Comment by gianlucadecola** [comment]
> Could you remove the TestingEnv, otherwise looks good from the discussions we had with Ariel. Two small points for going forward
> 
> 1. Don't worry testing vector implementations due to `VectorEnv` changes these will most likely have to be removed in the future.

Ok good

> 2. There are a couple of type hinting things we can improve (nothing breaking) however in an effect to merge this quickly I can do them after.

:+1:

File: tests/dev_wrappers/utils.py (hunk 3 above)
Comment (gianlucadecola) [reply tests/dev_wrappers/utils.py]: https://github.com/Farama-Foundation/Gymnasium/pull/148/commits/846676a1cd84fffdb48b64c48db69b1dafac1843

File: gymnasium/dev_wrappers/lambda_action.py (hunk 1 above)
Comment (gianlucadecola) [reply gymnasium/dev_wrappers/lambda_action.py]: https://github.com/Farama-Foundation/Gymnasium/pull/148/commits/0179f6666649d0f6b4d2319bfe75ce5f477f11b6

File: gymnasium/dev_wrappers/lambda_observations.py (hunk 2 above)
Comment (gianlucadecola) [reply gymnasium/dev_wrappers/lambda_observations.py]: https://github.com/Farama-Foundation/Gymnasium/pull/148/commits/75ba82082a91c2d5fbe2a4827139fffd8132fd23

**Review by pseudo-rnd-thoughts — APPROVED** [review]

**Pull request merged**